### ISSUE TITLE
Proposal for a new Routing system.

### ISSUE THREAD

**Issue by Deiru2k** [hfaran/Tornado-JSON#45]
I have been trying out the automatic routing system that you currently use in project, and I have found it to be not as convenient as I think it should be.
While it certainly is suited for some cases (i.e. when you store all your views inside one file, say, api.py), I found out that it doesn't work well with the project layout I find most convenient (and I have good reasons for that).
Currently, it walks a package for modules and creates routes respectively, so
`handlers.api.ImagesHandler`
becomes 
`/api/images`
But when working with larger projects, I find this to be not as convenient as it might seem. The way I use my project, is that I create a handlers package and create modules for each resource I use respectively, so in my case with your current auto-routing, the module will be
`handlers.images.ImagesHandler` and `handlers.images.ImageHandler`
Which would be mapped to 
`/images/images` and `/images/image`.
What I propose is a system that will A - have a defined base path for routes that can be defined when calling get_routes (i.e. get_routes(package, base='api'))
And names will be generated using the following scheme
`handlers.images.ImagesHandler -> /api/images`
`handlers.images.ImageHandler -> /api/images/(.* )`
With any additional loose handlers that don't contain module name within them will become actions for collection, like 
`handlers.images.PopularHandler -> /api/images/popular`
With exceptions and custom routes that can be defined inside Handler classes.

This system is more suited for bigger projects where having a separate module for each resource can be pretty convenient and makes it easier to navigate the project.

I understand that you might be focused on other things, or don't really feel like implementing such system, so I've decided to take it up on my own and fill a pull-request once I'm done.
I would also try to implement changes I've proposed in my previous issue regarding metadata in jsend.

**Comment by auvipy** [comment]
how about werkzeug like routing?

**Comment by auvipy** [comment]
http://truemped.github.io/tornadotools/route.html

**Comment by auvipy** [comment]
http://tornado-smack.readthedocs.org/en/latest/

**Comment by hfaran** [comment]
@auvipy Yes! This is along the lines of what I had in mind: https://github.com/hfaran/Tornado-JSON/compare/routing_refactor#diff-282bf316c179404e83b2e5db1ce3170bR256

I have not unfortunately had time to touch this in a while though because I've been quite busy recently but I hope to flesh this out more and finish it soon.

**Comment by hfaran** [comment]
https://github.com/troolee/tornado-routes
